File: PYTHON/TP2/test/test_etudiant.py
from src.etudiant import Etudiant
from src.cours import Cours

def test_creation_etudiant():
    maths = Cours("Mathématiques", "M.Dupond")
    français = Cours("Français", "M.Leroy")
    cours = [maths, français]
    etudiant = Etudiant("Bob", 22, "E12345", 15.5, cours)
    assert etudiant.get_nom() == "Bob"
    assert etudiant.get_age() == 22
    assert etudiant.numero_etudiant == "E12345"
    assert etudiant.get_moyenne() == 15.5
    assert etudiant.cours == cours

File: JAVA/TP2/test/TestEtudiant.java
package test;
import src.Personne;
import src.Etudiant;
import src.Cours;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.List;

public class TestEtudiant {

    @Test
    void test_creation_etudiant() {

        Cours maths = new Cours("Mathématiques", "M.Dupond");
        Cours francais = new Cours("Français", "M.Leroy");

        List<Cours> cours = List.of(maths, francais);

        Etudiant etudiant = new Etudiant("Bob", 22, 12345, 15.5);

        etudiant.ajouterCours(maths);
        etudiant.ajouterCours(francais);

        assertEquals("Bob", etudiant.getNom());
        assertEquals(22, etudiant.getAge());
        assertEquals(12345, etudiant.getNumeroEtudiant());
        assertEquals(15.5, etudiant.getMoyenne());
        assertEquals(2, etudiant.getCours().size());
    }
}